### PR TITLE
Update hetzner CCM to v1.17.1

### PR DESCRIPTION
**Pull request by kron4eg**
**Which issue(s) this PR fixes**:
Xref #2782

**What type of PR is this?**
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Update hetzner CCM to v1.17.1
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: addons/ccm-hetzner/ccm-hetzner.yaml [hunk 1]
```diff
@@ -31,15 +31,17 @@ spec:
   revisionHistoryLimit: 2
   selector:
     matchLabels:
-      app: hcloud-cloud-controller-manager
+      app.kubernetes.io/instance: 'hccm'
+      app.kubernetes.io/name: 'hcloud-cloud-controller-manager'
```
Comment (xmudrii) [inline addons/ccm-hetzner/ccm-hetzner.yaml:35]: This requires migration because PodSelector labels are immutable. See #2824

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: addons/ccm-hetzner/ccm-hetzner.yaml (hunk 1 above)
Comment (kron4eg) [reply addons/ccm-hetzner/ccm-hetzner.yaml:35]: Added migration path

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**